sycl/include/CL/sycl/event.hpp

```diff
@@ -132,16 +132,32 @@ class __SYCL_EXPORT event {
   /// \return a native handle, the type of which defined by the backend.
   template <backend BackendName>
   __SYCL_DEPRECATED("Use SYCL 2020 sycl::get_native free function")
-  auto get_native() const -> typename interop<BackendName, event>::type {
+  detail::enable_if_t<(BackendName != backend::opencl),
+                      typename interop<BackendName, event>::type> get_native()
+      const {
     return reinterpret_cast<typename interop<BackendName, event>::type>(
         getNative());
   }
+  ///
+  template <backend BackendName>
+  detail::enable_if_t<(BackendName == backend::opencl), std::vector<cl_event>>
+  get_native() const {
+    backend_return_t<BackendName, event> ReturnValue;
+    for (auto const &element : getNativeVector()) {
+      ReturnValue.push_back(
+          reinterpret_cast<typename interop<BackendName, event>::value_type>(
+              element));
+    }
+    return ReturnValue;
+  }
```

Comment:
Why do we need this function at all?

Reply:
According to [SYCL 2020 specification (revision 3)](https://www.khronos.org/registry/SYCL/specs/sycl-2020/pdf/sycl-2020.pdf#page=478&zoom=100,64,364) get_native<sycl::backend::opencl>(event) should return vector<cl_event>, whereas for others backends it is not specified. Because C++14 has no "if constexpr", I chose to change SFINAE.

Reply:
event::get_native is deprecated. We should not change it in any way.

Reply:
At this moment, sycl::get_native free function calls a deprecated function with backend=opencl, so to change the free function we need to work with deprecated.

Reply:
I think Alex is right - I'm not even sure where this public function event::get_native() came from.

Regardless, since it's an existing function with different behavior, we should leave it as-is to avoid the breaking change.

I think we need to change the impl of our free function to not use this deprecated method.

Reply:
@jbrodman is it ok that with any changes API will break, because backend_return_t for opencl event will return std::vector<cl_event> instead of cl_event?